onnxruntime/test/providers/cpu/controlflow/scan_test.cc

```diff
@@ -334,10 +339,13 @@ static void RunTest_v9(const std::string test_name, int64_t sequence_len, int64_
   // create model that will be used to initialize subgraph. currently there's no direct way to create a Graph instance.
   Model model(test_name, false, DefaultLoggingManager().DefaultLogger());
   auto& graph = model.MainGraph();
-  CreateSubgraph(graph, options, options.add_bad_shape ? failure_message : "");
+  auto status = CreateSubgraph(graph, options, options.add_bad_shape ? failure_message : "");
+  if (!status.IsOK()) {
+    return;
+  }
```

Comment:
does this test pass if CreateSubgraph fails?

Reply:
No, it won't pass. it would stop here. is that a concern?

Reply:
This is the reason for fixing BadShape test.

Reply:
yes that makes sense, when you use the latest version the test infrastructure breaks immediately for strict checking but with an older version it swallows some exception but you are masking that returning here so I think the intended code paths are being tested.  

---
In reply to: [395815076](https://github.com/microsoft/onnxruntime/pull/3278#discussion_r395815076) [](ancestors = 395815076)

Reply:
correct, the older logic swallowed exception indeed.